src/components/capture/DetailsForm.tsx

```diff
@@ -29,13 +31,29 @@ const stateSelectors: StoreSelector<CaptureCreationState> = {
 
 function NewCaptureDetails({ connectorTags }: Props) {
     const intl = useIntl();
+    const [searchParams] = useSearchParams();
+    const connectorID = searchParams.get(
+        routeDetails.capture.create.params.connectorID
+    );
+
     const formData = useCaptureCreationStore(stateSelectors.formData);
     const setDetails = useCaptureCreationStore(stateSelectors.setDetails);
     const displayValidation = useCaptureCreationStore(
         stateSelectors.showValidation
     );
     const status = useCaptureCreationStore(stateSelectors.status);
 
+    useEffect(() => {
+        if (connectorID) {
+            setDetails({
+                data: {
+                    name: '',
+                    image: connectorID,
+                },
+            });
+        }
+    }, [connectorID, setDetails]);
```

Comment:
Initially I put this into the index file. However, it feels better in this component. Mainly because this is the component that "owns" updating details... so I think it makes sense here.